src/Components/Components/src/DynamicComponent.cs

```diff
@@ -32,6 +32,7 @@ public DynamicComponent()
         /// </summary>
         [Parameter]
         [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)]
+        [EditorRequired]
```

Comment:
It's already required [here](https://github.com/dotnet/aspnetcore/pull/33384/files#diff-890a8f579916b14b8e7801cb9525e835e4d7d4321efe5a88ef8d78621620df92L80-L83)

Reply:
If you are asking for `[Required]` that's not something Blazor can support since setting parameters is `virtual` and can easily be bypassed, so there's no way to offer that guarantee.

Reply:
Our initial implementation in 5.0 tried a runtime check, but we quickly found out that it does not work all the time. Having it be unreliable makes it a no-go.